README.md

```diff
@@ -1,11 +1,158 @@
+
+# Special cases and considerations with Periods logic
+
+There are some special DHIS-2 specific cases when doing period calcualtions that we handle. The specs for these are documented in the tests and cucumber feature files. Some of these are:
+
+- Dealing with the 13th month in the Ethiopian calendar: the Ethiopic calendar has 13 months: 12 months of 30 days, then the 13th month has 5 or 6 days depending on whether it's a leap year or not. When generating periods for bi-weekly or less (bi-weekly, weekly, daily) then the 13th month is displayed. But when we are generating periods for Monthly or larger, we do not display the 13th month. The backend then makes the decision about where to include any data for the 13th month, so when doing monthly analytics, the 13th month will be lumped with the following month's data.
+
+- The period IDs are dictated by the Backend, and they use the gregorian month names, for example, `QuarterlyNov` or `QuarterlyApr`. In the frontend, in a non-Gregorian calendar, `Nov` will be "translated" into the 11th month of the current calendar system (Hamle in Ethiopic). Ideally, we will have Period IDs that are calendar-agnostic in the future, but this works for now as a solution.
+
+- From Frontend perspective, we always send a date in the calendar system. It is formatted in an iso-like format (`yyyy-MM-dd`). It is "iso-like", because the date is actually in the calendar system rather than the gregorian iso8601 calendar. So when a system set to Ethiopic, sends `2015-01-01` to the Backend, it means the year 2015 in Ethiopic (which is 2022 in Gregorian). This is what the Backend expects, and if there is a need for a conversion, then it happens in the backend.
+  - Given we have the Temporal object in the frontend, we have the ability to send the date converted into Gregorian, but that's what the backend expects right now.
+
+- Nepali is implemented as a custom calendar (https://github.com/dhis2/multi-calendar-dates/blob/multi-calendar-docs/src/custom-calendars/nepaliCalendar.ts). Nepali is luni-solar calendar which means that the length of months can vary from year to year. The way Temporal does date arithmetic, is that it converts the day to gregorian, performs the arithmetic, then converts it back to the calendar. This leads to weird situations where you do `nepaliDate.add({month: 1})` for example, but the result is still in the current month (since the month has 30 days in Gregorian, but 32 in Nepali for example). To get around that, we [set the day to the 14th](https://github.com/dhis2/multi-calendar-dates/blob/73057360c4720d995370779b02fe6e3784b326ab/src/period-calculation/getYearlyPeriods.ts#L23) before doing arithmetic (in the context of period generation) and this leads to results that make sense in the context of DHIS2.
```

Comment:
```suggestion
- Nepali is implemented as a custom calendar: see [nepaliCalendar.ts](https://github.com/dhis2/multi-calendar-dates/blob/multi-calendar-docs/src/custom-calendars/nepaliCalendar.ts). Nepali is luni-solar calendar which means that the length of months can vary from year to year. The way Temporal does date arithmetic, is that it converts the day to gregorian, performs the arithmetic, then converts it back to the calendar. This leads to weird situations where you do `nepaliDate.add({month: 1})` for example, but the result is still in the current month (since the month has 30 days in Gregorian, but 32 in Nepali for example). To get around that, we [set the day to the 14th](https://github.com/dhis2/multi-calendar-dates/blob/73057360c4720d995370779b02fe6e3784b326ab/src/period-calculation/getYearlyPeriods.ts#L23) before doing arithmetic (in the context of period generation) and this leads to results that make sense in the context of DHIS2.
```
Just a suggestion for a shorter link